### PR TITLE
Using accreal instead of real in the API

### PR DESCRIPTION
**Pull request by pavanky**
This is done to be consistent with the changes made to cunn

### CODE REVIEW

**Comment by pavanky** [comment]
This PR depends on: https://github.com/torch/torch7/pull/915

File: THNN.lua [hunk 1]
```diff
@@ -73,6 +73,13 @@ local replacements =
     }
 }
 
+-- gsub(s, 'real', 'float') changes accreal to accfloat.
+-- typedef accfloat ahead of time.
+ffi.cdef("typedef double accfloat;")
```
Comment (gchanan) [inline THNN.lua]: shouldn't accfloat be float?

Comment (pavanky) [inline THNN.lua]: It is `double` in `THGenerateFloatType.h`

Comment (gchanan) [inline THNN.lua]: ah you are correct, didn't realize they were different on cpu vs gpu.

Comment (pavanky) [inline THNN.lua]: But it is `float` in the CUDA version for performance reasons. I talked to soumith about it and decided to not change the way they are right now.

**Pull request merged**